console/collections/src/kary_merkle_tree/helpers/leaf_hash.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use snarkvm_console_algorithms::{Keccak, Poseidon, BHP};
+use snarkvm_console_types::prelude::*;
+
+use crate::kary_merkle_tree::BooleanHash;
+#[cfg(not(feature = "serial"))]
+use rayon::prelude::*;
+
+/// A trait for a Merkle leaf hash function.
+pub trait LeafHash: Clone + Send + Sync {
+    type Hash: Copy + Clone + Debug + Default + PartialEq + Eq + FromBytes + ToBytes + Send + Sync;
+    type Leaf: Clone + Send + Sync;
+
+    /// Returns the hash of the given leaf node.
+    fn hash_leaf(&self, leaf: &Self::Leaf) -> Result<Self::Hash>;
+
+    /// Returns the hash for each leaf node.
+    fn hash_leaves(&self, leaves: &[Self::Leaf]) -> Result<Vec<Self::Hash>> {
+        match leaves.len() {
+            0 => Ok(vec![]),
+            1..=100 => leaves.iter().map(|leaf| self.hash_leaf(leaf)).collect(),
```

Comment:
IIRC the value of `100` was picked empirically for the Merkle tree; we might want to check if this is the right pick for this type of tree too (i.e. run benchmarks with some alternatives)

Reply:
I think it remains the same given 2-ary is also supported here.